src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrForwardFlower.cs

```diff
@@ -558,19 +558,12 @@ private async Task<bool> RecreatePreviousFlowAndApplyBuild(
     {
         _logger.LogInformation("Failed to create PR branch because of a conflict. Re-creating the previous flow..");
 
-        // Find the BarID of the last flown repo build
-        RepositoryRecord previouslyAppliedRepositoryRecord = _sourceManifest.GetRepositoryRecord(mapping.Name);
-        Build previouslyAppliedBuild;
-        if (previouslyAppliedRepositoryRecord.BarId == null)
+        // Create a fake previously applied build. We only care about the sha here, because it will get overwritten anyway
+        Build previouslyAppliedBuild = new(-1, DateTimeOffset.Now, 0, false, false, lastFlow.SourceSha, [], [], [], [])
```

Comment:
`VmrBackflower` has `lastLastFlow.SourceSha` here instead. Should that be here as well?